blueocean-web/src/main/js/main.jsx

```diff
@@ -37,17 +39,18 @@ class App extends Component {
     }
 
     render() {
+        const { t } = this.props;
```

Comment:
we are passing functionality to the component as you do in react. Why should we not use a plugin such as react-i18next for the underlying i18next provider. 

http://i18next.com/docs/api/

In case you need it outside the react way see above api, but in React components the above is the way to go IMHO